assembl/static2/js/app/pages/debate.jsx

```diff
@@ -30,11 +34,41 @@ class Debate extends React.Component {
   displayThumbnails() {
     this.setState({ isThumbnailsHidden: !this.state.isThumbnailsHidden });
   }
+  redirectToV1Threads() {
+    const { debateData } = this.props.debate;
+    const { identifier } = this.props;
+    const locale = this.props.lang;
+    const phaseName = getPhaseName(debateData.timeline, identifier, locale);
+    const body = <Translate value="debate.redirectToThreadPhase" phaseName={phaseName} />;
+    displayModal(null, body, true, null, null, true);
+    const slug = { slug: debateData.slug };
+    const redirectionURL = get('oldDebate', slug);
+    setTimeout(() => {
+      location.href = redirectionURL;
+    }, 6000);
+  }
   render() {
+    const { debateData } = this.props.debate;
+    const { identifier } = this.props; // identifier of current phase being accessed
     const { loading, thematics } = this.props.data;
-    const { identifier, isNavbarHidden } = this.props;
+    const { isNavbarHidden } = this.props;
     const isParentRoute = !this.props.params.phase || false;
     const themeId = this.props.params.themeId || null;
+
+    if (identifier && !isPhaseStarted(debateData.timeline, identifier)) {
```

Comment:
Hmm.. `isPhaseStarted` is not a promise. Therefore, my suggestion is to take this entire `if/else` block, and put it in the `componentDidMount`. This method is called after render. And it seems that there is no `setState` called in your block, so no re-render will be done. Safe.